contrib/boot-contracts-stateful-prop-tests/tests/pox-4/pox-4.stateful-prop.test.ts

```diff
@@ -143,17 +145,22 @@ it("statefully interacts with PoX-4", async () => {
 
   simnet.setEpoch("3.0");
 
+  const successPath = PoxCommands(model.wallets, model.stackers, sut.network);
+  const failurePath = ErrCommands(model.wallets, model.stackers, sut.network);
```

Comment:
This PR adds a set of stateful property-tests for the error/failure path(s) in PoX-4. Reviewers should check [contrib/boot-contracts-stateful-prop-tests/tests/pox-4/err_Commands.ts](https://github.com/stacks-network/stacks-core/pull/4842/files#diff-c0b7d803fd3422af995be6248a171114c91771ad7e11ec7b18d92b0db775e270).

The rest of the files in this PR, apart from contrib/boot-contracts-stateful-prop-tests/tests/pox-4/err_Commands.ts are infrastructure.